src/components/MenuIcon.tsx

```diff
@@ -0,0 +1,20 @@
+import React, {Dispatch, SetStateAction} from 'react';
+
+interface MenuIconProps {
+  setToggleMenu: Dispatch<SetStateAction<boolean>>;
+}
+
+const MenuIcon = ({setToggleMenu}: MenuIconProps) => {
+  const openMenu = () => {
```

Comment:
declare openMenu function in the parent component and pass it as props, try not to pass setState function as props